src/configure/helper/templateHelper.ts

```diff
@@ -112,35 +139,50 @@ export async function analyzeRepoAndListAppropriatePipeline(repoPath: string, re
     templateResult = templateResult.filter((pipelineTemplate) => pipelineTemplate.enabled);
 
-    // remove duplicate named template:
-    templateResult = removeDuplicates(templateResult);
     return templateResult;
-
 }
 
-export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplateMetadata[]> {
+export async function analyzeRepoAndListAppropriatePipeline2(azureSession: AzureSession, repoPath: string, repositoryProvider: RepositoryProvider, repoAnalysisParameters: RepositoryAnalysisParameters, targetResource?: GenericResource): Promise<PipelineTemplate[]> {
 
-    //TO:DO - Merge local repo analysis (Some changes in the definition of AnalysisResult required)
-    let templateResult: PipelineTemplateMetadata[] = [];
+    var pipelineTemplates: PipelineTemplate[] = [];
+    var localPipelineTemplatesPromise: Promise<LocalPipelineTemplate> = await this.analyzeRepoAndListAppropriatePipeline(repoPath, repositoryProvider, repoAnalysisParameters);
 
     let serviceClient = new TemplateServiceClient(azureSession.credentials);
-    templateResult = await serviceClient.getTemplates(repoAnalysisParameters);
-    templateResult = templateResult.sort((a, b) => {
-        if (a.templateWeight > b.templateWeight) { return 1; }
-        else { return -1; }
-    });
-    return templateResult;
+    var templateResultPromise = await serviceClient.getTemplates(repoAnalysisParameters);
+
+    return Q.all([localPipelineTemplatesPromise, templateResultPromise])
+        .spread((localPipelineTemplates: LocalPipelineTemplate, templateResult: PipelineTemplateMetadata[]) => {
+            templateResult.forEach((template: PipelineTemplateMetadata) => {
+                var remoteTemplate: RemotePipelineTemplate = {
+                    label: template.templateDescription,
+                    targetType: getTargetType(template),
+                    targetKind: getTargetKind(template),
+                    language: template.attributes.language,
+                    templateId: template.templateId,
+                    templateWeight: template.templateWeight,
+                    templateType: TemplateType.remote,
+                    workingDirectory: template.workingDirectory
+                };
+                pipelineTemplates.push(remoteTemplate);
+            });
+            // sorted by weight
+            pipelineTemplates = pipelineTemplates.sort((a, b) => {
+                if (a.templateWeight < b.templateWeight) { return 1; }
+                else { return -1; }
+            });
+            return pipelineTemplates.concat(localPipelineTemplates);
```

Comment:
Should we not filter out local templates if they  have same name?

Reply:
At this stage, we aren't sure whether we'll proceed with local or remote template, as the user hasn't selected the webApp yet. So we are storing the potential templates as :

```
this.inputs.potentialTemplates = pipelineMap.get(selectedOption.label);
```

Once the user selects the webApp, will filter out the tenplates:

```
shortlistedTemplates = this.inputs.potentialTemplates.filter((template) => template.targetKind === resource.kind);
                if (shortlistedTemplates.length > 1) {
                    this.inputs.pipelineConfiguration.template = shortlistedTemplates.find((template) => template.templateType === TemplateType.REMOTE);
                }
```

Reply:
We will always use remote template over a local template with same name. This fact does not change even if specific template is selected only after webapp is selected

Reply:
We can't filter local templates even if the name is same at this stage because local template can be for different kind of web App like windows and remote templates may not contain windows webapp tempates. We will get to know the webAppKind after the webApp selection.